Sources/FoundationEssentials/UUID.swift

```diff
@@ -76,6 +76,50 @@ public struct UUID : Hashable, Equatable, CustomStringConvertible, Sendable {
             hasher.combine(bytes: buffer)
         }
     }
+    
+    /// Generates a new random UUID.
+    ///
+    /// - Parameter generator: The random number generator to use when creating the new random value.
+    /// - Returns: A random UUID.
+    @available(FoundationPreview 6.3, *)
+    public static func random(
+        using generator: inout some RandomNumberGenerator
+    ) -> UUID {
+        let first = UInt64.random(in: .min ... .max, using: &generator)
+        let second = UInt64.random(in: .min ... .max, using: &generator)
+
+        var firstBits = first
+        var secondBits = second
+
+        // Set the version to 4 (0100 in binary)
+        firstBits &= 0xFFFFFFFFFFFF0FFF // Clear bits 48 through 51
```

Comment:
Should we use binary literals (`0b....`), that way it's much easier to read? (longer ofc)